web3-react/src/libs/connections.ts

```
@@ -0,0 +1,109 @@
+  [ConnectionType.NETWORK]: buildNetworkConnector(),
+}
+
+export function getConnection(c: Connector | ConnectionType) {
+  if (c instanceof Connector) {
+    const connection = Object.values(PRIORITIZED_CONNECTORS).find((connection) => connection.connector === c)
+    if (!connection) {
+      throw Error('unsupported Connector')
+    }
+    return connection
+  } else {
+    switch (c) {
+      case ConnectionType.INJECTED:
+        return PRIORITIZED_CONNECTORS[ConnectionType.INJECTED]
+      case ConnectionType.COINBASE_WALLET:
+        return PRIORITIZED_CONNECTORS[ConnectionType.COINBASE_WALLET]
+      case ConnectionType.WALLET_CONNECT:
+        return PRIORITIZED_CONNECTORS[ConnectionType.WALLET_CONNECT]
+      case ConnectionType.GNOSIS_SAFE:
+        return PRIORITIZED_CONNECTORS[ConnectionType.GNOSIS_SAFE]
+      case ConnectionType.NETWORK:
+        return PRIORITIZED_CONNECTORS[ConnectionType.NETWORK]
+    }
+  }
+}
+
+export const switchNetwork = async (chainId: number, connectionType: ConnectionType | null) => {
+  if (!connectionType) {
+    return
+  }
+
+  const { connector } = getConnection(connectionType)
+
+  if (connectionType === ConnectionType.WALLET_CONNECT || connectionType === ConnectionType.NETWORK) {
+    await connector.activate(chainId)
+    return
+  }
+
+  const chainInfo = CHAIN_INFO[chainId]
+  const addChainParameter: AddEthereumChainParameter = {
+    chainId,
+    chainName: chainInfo.label,
+    rpcUrls: [chainInfo.rpcUrl],
+    nativeCurrency: chainInfo.nativeCurrency,
+    blockExplorerUrls: [chainInfo.explorer],
+  }
+  await connector.activate(addChainParameter)
+}
+
+// Try to activate a connector
+export const tryActivateConnector = async (connector: Connector): Promise<ConnectionType | undefined> => {
+  try {
+    await connector.activate()
+    const connectionType = getConnection(connector).type
+    return connectionType
+  } catch (error) {
+    console.debug(`web3-react connection error: ${error}`)
```

Is it expected that we swallow errors? I would personally prefer us to have a high-level and simple `ErrorBoundary` that we can use to present a nice error, or anything in the UI. While this is a simple example, I think we should showcase some best practices when it comes to using `web3-react`.